src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool/CompareBinariesAgainstBaselines.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.Logging;
+
+namespace BinaryTool;
+public partial class Driver
+{
+    private const string PatternsToRemoveMarker = "# Consider removing the following unused patterns from the baseline file:";
+
+    // Requires: DetectedBinariesFile exists
+    // Modifies: UpdatedAllowedBinariesKeepFile, UpdatedAllowedBinariesRemoveFile, NewBinariesFile, RemovedBinariesFile
+    // Effects:  Compares the detected binaries against the baseline patterns and writes the results to the output files.
+    private void CompareBinariesAgainstBaselines()
+    {
+        Log.LogInformation($"Validating binaries in {TargetDirectory} {GetBaselineLogString()}...");
+
+        IEnumerable<string> detectedBinaries = File.ReadLines(DetectedBinariesFile);
```

Comment:
The design of `Driver` seems very file-based when it doesn't need to be. It's writing to files as output and then reading those files for other operations. It should just pass the content directly as strings rather than using the file system.

Reply:
This should be blazingly fast.  The file writing is adding a lot of overhead that could be avoided.  I think it would be best to avoid the intermediate files.

Reply:
How important is the updating of the baseline to alert to unused patterns? Without that, I can exclude the work to update the baseline files which would improve the speed and clarify of this tool.

Reply:
Another thing to consider is if we want to separate out the tasks in this tool.

Currently, the tasks are chained together as detect -> validate -> clean. We can restructure this into two separate processes: detect -> clean and detect -> validate. This would allow the user to choose either detect -> clean, detect -> validate, or the entire detect -> validate -> clean sequence.

Another alternative is to design the tool only for detection and cleaning, and shift validation to a smoke test. This approach incurs more repeated code overhead due to the necessary detection component for both cleaning and validation, but it would emphasize the tool's main objective: to clean/remove binaries.

Reply:
The only files that are currently written to are the updated baselines, and that writing only occurs when validation is being run. All other file writing has been removed.